puma/histogram.py

```diff
@@ -432,14 +455,30 @@ def plot(self, **kwargs):
         for key in self.add_order:
             elem = self.plot_objects[key]
 
-            elem.bin_edges, elem.hist, elem.unc, elem.band = hist_w_unc(
-                elem.values,
-                weights=elem.weights,
-                bins=self.bins,
-                bins_range=self.bins_range,
-                normed=self.norm,
-                underoverflow=self.underoverflow,
-            )
+            # if the histogram is not filled then deal with it normally
+            if not elem.filled: 
```

Comment:
See my comment in the `hist_w_unc` function below but if these two functions are merged we can use the same function call here